### PR TITLE
[python] [bug] Fixed handling of dotted module names

### PR DESCRIPTION
**Pull request by svenpanne**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. (Irrelevant point, Travis CI checked the PR in my fork.)
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Previously, if you used a packageName of the form `foo.bar.baz`, half of the
generated files of the python generator went into a subdirectory
`foo/bar/baz` (correct), the other half went into a subdirectory
`foo.bar.baz` (incorrect).

@taxpon @frol @mbohlool @cbornet @kenjones-cisco @tomplus @Jyhess

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java [hunk 1]
```diff
@@ -201,12 +201,14 @@ public void processOpts() {
         additionalProperties.put(CodegenConstants.PACKAGE_NAME, packageName);
         additionalProperties.put(CodegenConstants.PACKAGE_VERSION, packageVersion);
 
+        // NOTE: The null case can happen during PythonClientOptionsTest.checkOptionsProcessing().
+        String packagePath = packageName == null ? "" : packageName.replace('.', File.separatorChar);
```
Comment (rienafairefr) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java]: Would be cleaner to have packageName != null during the unit test `PythonClientOptionsTest.checkOptionsProcessing` ?

Comment (svenpanne) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java]: Yes, that would be cleaner, but due to the slightly obscure mocking Kung Fu in the test I was unable to come up with something better.

I propose to land this change as it is (perhaps add a TODO), the current module name handling is totally unusable in non-trivial existing Python projects. Alternatively, somebody could give me a concrete hint how to fix the test... :wink:

**Comment by rienafairefr** [comment]
Tested on a real spec & project, works fine :+1:

**Comment by wing328** [comment]
@svenpanne thanks for the PR. What about using the `packagePath()` function approach in https://github.com/OpenAPITools/openapi-generator/pull/2041 ?

Later we'll create a Python abstract class for Python client and server generators so keeping things a bit more consistent will make the refactoring easier.

**Comment by svenpanne** [comment]
> @svenpanne thanks for the PR. What about using the `packagePath()` function approach in #2041 ? [...]

Good idea, done. Note that I've fixed and improved the corresponding options test a bit, so the funny null check is gone. `jmockit` is a strange beast if you don't use it on a daily basis... :confused:

**Pull request merged**